homeassistant/components/tellduslive.py

```diff
@@ -45,58 +52,178 @@
+click the link, login, and authorize {app_name}.
+Then click the Confirm button.
+"""
 
-def setup(hass, config):
+
+def setup(hass, config, session=None):
     """Set up the Telldus Live component."""
-    client = TelldusLiveClient(hass, config)
+    from tellduslive import Session, supports_local_api
+    config_filename = hass.config.path(TELLLDUS_CONFIG_FILE)
+    conf = load_json(config_filename)
+
+    def request_configuration(host=None):
+        """Request TelldusLive authorization."""
+        configurator = hass.components.configurator
+        hass.data.setdefault(KEY_CONFIG, {})
+        data_key = host or DOMAIN
+
+        # Configuration already in progress
+        if hass.data[KEY_CONFIG].get(data_key):
+            return
+
+        _LOGGER.info('Configuring TelldusLive %s',
+                     'local client: {}'.format(host) if host else
+                     'cloud service')
+
+        session = Session(public_key=PUBLIC_KEY,
+                          private_key=NOT_SO_PRIVATE_KEY,
+                          host=host,
+                          application=APPLICATION_NAME)
+
+        auth_url = session.authorize_url
+        if not auth_url:
+            _LOGGER.warning('Failed to retrieve authorization URL')
+            return
+
+        _LOGGER.debug('Got authorization URL %s', auth_url)
+
+        def configuration_callback(callback_data):
+            """Handle the submitted configuration."""
+            session.authorize()
+            res = setup(hass, config, session)
+            if not res:
+                hass.async_add_job(configurator.notify_errors,
+                                   hass.data[KEY_CONFIG].get(data_key),
+                                   'Unable to connect.')
+                return
+
+            @asyncio.coroutine
+            def success():
+                """Set up was successful."""
+                conf.update(
+                    {host: {CONF_HOST: host,
+                            CONF_TOKEN: session.access_token}} if host else
+                    {DOMAIN: {CONF_TOKEN: session.access_token,
+                              CONF_TOKEN_SECRET: session.access_token_secret}})
+                save_json(config_filename, conf)
+                # Close all open configurators: for now, we only support one
+                # tellstick device, and configuration via either cloud service
+                # or via local API, not both at the same time
+                for instance in hass.data[KEY_CONFIG].values():
+                    hass.async_add_job(configurator.request_done,
+                                       instance)
+
+            hass.async_add_job(success)
+
+        hass.data[KEY_CONFIG][data_key] = \
+            configurator.request_config(
+                'TelldusLive ({})'.format(
+                    'LocalAPI' if host
+                    else 'Cloud service'),
+                configuration_callback,
+                description=CONFIG_INSTRUCTIONS.format(
```

Comment:
The description can now be markdown. So you can directly embed images and links. The `link_name` and `link_url` are deprecated.

Reply:
This is addressed at: https://github.com/home-assistant/home-assistant/pull/10435/commits/af44a5b20a971833ed5ec7eaff6cfabbdf9d70ef#R54